Drop Python 3.5 support

jgosmann opened the pull request:
**Motivation and context:**
<!--- Why is this change required? What problem does it solve? -->
<!--- If it addresses an open issue, please link to the issue here. -->

Drop support for Python 3.5 because nengo-bones does not work with it anymore.

* Change string interpolations and string `format()` calls to f-strings which were introduced with Python 3.6. A few `format()` calls remain where an f-string either cannot be used or would be less legible.

**Interactions with other PRs:**
<!--- If this change depends on or conflicts with another PR please list it here. -->
<!--- If this PR contains commits from another PR, describe what commits in this PR are unique. -->
<!--- If this PR is independent, then remove this section. -->
Fixes build for all other PRs. Other PRs should switch over to f-strings.

**How has this been tested?**
<!--- Please describe in detail how you tested your changes. -->
<!--- Reviewers will test your PR in at least this way. -->
existing tests

**How long should this take to review?**
<!--- Please estimate if this PR is a quick, average, or lengthy PR. -->
<!--- Take into account both the size and complexity of the changes. -->
<!--- Also note if this is a timely PR that should be reviewed by a certain date. -->
<!--- Leave only the line that applies below: -->

- Average (neither quick nor lengthy)

**Where should a reviewer start?**
<!--- If the PR warrants it, indicate where a reviewer should start reviewing. -->
<!--- All lengthy PRs and complicated average PRs warrant this section. -->
<!--- If the PR is quick or straightforward, remove this section. -->
Changes to the .nengobones.yml might be a good place?

**Types of changes:**
<!--- What types of changes does your code introduce? -->
<!--- Leave all lines that apply below: -->

- Maintenance

**Checklist:**
<!--- Go over all the following points. Put an `x` in all the boxes that apply. -->
<!--- If a box is not applicable, please justify below the checklist. -->
<!--- If you're unsure about any of these, don't hesitate to ask. -->
<!--- We're here to help! -->

- [x] I have read the **CONTRIBUTING.rst** document.
- [x] I have updated the documentation accordingly.
- [x] I have included a changelog entry.
- [n/a] I have added tests to cover my changes.
- [x] I have run the test suite locally and all tests passed.

**Still to do:**
<!--- If this is a work in progress, note below what you still plan to do. -->
<!--- Use the task list syntax `- [ ]` so that progress can be tracked. -->

* [x] Double check that missing string interpolation arguments have been filled in correctly.

hunse commented on .nengobones.yml:
```diff
@@ -12,26 +12,27 @@ license_rst: {}
 
 travis_yml:
-  python: 3.7
+  python: 3.9
```
We don't officially support Python 3.9 yet in Nengo, though that's not to say it might not all work fine.

tbekolay replied:
Unofficially, I'm currently using Python 3.9, but with 3.8 the issue was Windows specific so who knows.

jgosmann replied:
I think it doesn't hurt to run the tests on 3.9 too as long as they pass. I also think that the "Python 3.6+" formulation in the the readme is fine (NengoSPA is more platform independent than Nengo, so it is even more unlikely that a new Python release breaks things). Might consider taking the classifier out of the setup.py ... but that I would have to re-add it later.

jgosmann commented:
Force pushed a few more changes:
* Added 16fb5a66ee33104554c92c781b2744e2fbc2f38e
* Amended b427ddbdd2a2fbaa9d6d486d0b972d76953ce369 to fix docs build
* Added commit to fix deprecated intersphinx links
